packages/cli/src/report/report-disk-writer.ts

```diff
@@ -10,13 +10,19 @@ import { ReportFormats } from './report-formats';
 export class ReportDiskWriter {
     constructor(private readonly fileSystemObj: typeof fs = fs) {}
 
-    public writeToDirectory(directory: string, fileName: string, format: ReportFormats, content: string): void {
+    public writeToDirectory(directory: string, fileName: string, format: ReportFormats, content: string): string {
         if (isEmpty(directory)) {
             // tslint:disable-next-line: no-parameter-reassignment
             directory = '.';
         }
 
-        const reportFileName = `${directory}/${filenamify(fileName, { replacement: '_' })}.${format}`;
+        let reportFileName;
+
+        try {
+            reportFileName = `${directory}/${filenamify(fileName, { replacement: '_' })}.${format}`;
+        } catch {
+            reportFileName = `${directory}/${fileName}.${format}`;
```

Comment:
why do we need this try catch

Reply:
if the file name has space, it throws exception!

Reply:
May be we need to replace it to with dash instead

Reply:
why?

Reply:
We are using a dedicated package to handle the conversion of a string to a file name and it clearly not working  as it should be. Either the package should convert correctly w/o throwing an error (we may need to adjust function call maybe?) or look for other package/solution. Error on a space is quite odd.

Reply:
yeah the issue isn't the underscore, and failing if a url has a space is expected, because we use 'filenamify-url' which expect url, and url shouldn't has a space, it should be encoded like '%20' or something, so I'll use 'filenamify-url' and if it fails, I'll use 'filenamify' which expect just a string